### PR TITLE
Saeb dev 20220225 Adobe Analytic to Azure python class

### PR DESCRIPTION
**Pull request by JuntaoZ-NC**
Previously closed PR due to duplicate commits :https://github.com/DTS-STN/CDP-Lessons-Learned/pull/11

This PR is meant to be the first PR to provide step-by-step tutorial for the AA-to-Azure-Python-Wrapper Class created.

Here are the items included in this PR:

Requirements file
sample codes
AA-to-Azure-Python-Wrapper Class itself
Readme files
1.General Readme that provides insights to the integration from AA to Azure finally Power BI
2.Azure specific Readme that shows step by step guidance related to Azure
2.PowerBI specific Readme that shows step by step guidance related to Power BI

Requirements file
requirements.txt file for python to install dependencies

sample codes
1.ADF sample pipeline
2.ADF pipeline to integrate notebook with other activities

AA-to-Azure-Python-Wrapper Class
analytics_client Class that is designed to be imported as a wrapper to request AA data

Signed-off-by: jz <juntao.zou@hrsdc-rhdcc.gc.ca>

### CODE REVIEW

File: Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+
+
+
+
+# Ingest data from Adobe Analytic using a local Python script
+
+
+[OnPrem Python script - JWT example](examples/OnPrem/OnPremExample.py)
```
Comment (s-laugh) [inline Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md:15]: This is a dead link

File: Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (s-laugh) [inline Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md:1]: Why is this md file needed if you have the python script? This doesn't seem to add any additional information?

File: Adobe-Azure-analytics-api-v2.0/examples/ADFpipeline/ADFpipelineExample.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+
+
+
+
+
+# Orchestrate Databrick Notebook Activity using ADF
+
+
+[ADF pipeline]https://github.com/DTS-STN/AP-DataFactory/blob/main/pipeline/sclabs_site_analytics.json
```
Comment (s-laugh) [inline Adobe-Azure-analytics-api-v2.0/examples/ADFpipeline/ADFpipelineExample.md:15]: Should actually be a link
```suggestion
[ADF pipeline](https://github.com/DTS-STN/AP-DataFactory/blob/main/pipeline/sclabs_site_analytics.json)
```

**Review by s-laugh — CHANGES_REQUESTED** [review]
There are a few things I believe need some fixing (noted in comments) including
- to `.png`s aren't loading `adobe-analytics-client-id-secret-v2.png` and `adobe-analytics-generate-pub-priv-keys.png`
- Your commits still need to be signed with a GPG key.

File: Adobe-Azure-analytics-api-v2.0/src/analytics/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+from src.analytics.AnalyticsClient import analytics_client
```
[the remark this answers is not in the document]
Comment (JuntaoZ-NC) [reply Adobe-Azure-analytics-api-v2.0/src/analytics/__init__.py]: Deleted the Init file as the direct import line existed in example python scripts 
https://github.com/DTS-STN/CDP-Lessons-Learned/pull/13/commits/569e096850596ca74dcdf05c15226824da13022d

File: Adobe-Azure-analytics-api-v2.0/Azurepipeline.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+
+2.Follow the instructions below:  
+![](/assets/images/AA-to-Azure-Python-Wrapper-Class/Azure-Data-Factory-access-pipeline.png)
+
+3.Access to the Databrick activity:
+![](/assets/images/AA-to-Azure-Python-Wrapper-Class/Azure-Databricks-open.png)
+
+
+4.View the Databrick Notebook with the following content: [AnalyticsClient - Cloud(Azure) Class](examples/DatabricksNotebook/DatabricksExample.ipynb)
```
[the remark this answers is not in the document]
Comment (JuntaoZ-NC) [reply Adobe-Azure-analytics-api-v2.0/Azurepipeline.md]: modified Databricks code sample link to AP-Databricks 
https://github.com/DTS-STN/CDP-Lessons-Learned/pull/13/commits/c4bd83826fa7644fe10e2eb1b26159c756788338

File: Adobe-Azure-analytics-api-v2.0/examples/ADFpipeline/ADFpipelineExample.md (hunk 3 above)
Comment (JuntaoZ-NC) [reply Adobe-Azure-analytics-api-v2.0/examples/ADFpipeline/ADFpipelineExample.md]: In this Commit, examples subdir has been modified :

- Consolidated Databricks & ADF examples and Readme into "Azure" dir
- Consolidated On Prem Readme and script into "OnPrem" dir

https://github.com/DTS-STN/CDP-Lessons-Learned/pull/13/commits/4c70f5d422ebb4ec01594cf66c3fa759b70ff04d

File: Adobe-Azure-analytics-api-v2.0/examples/ADFpipeline/ADFpipelineExample.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+
```
[the remark this answers is not in the document]
Comment (JuntaoZ-NC) [reply Adobe-Azure-analytics-api-v2.0/examples/ADFpipeline/ADFpipelineExample.md]: In this Commit, examples subdir has been modified :

- Consolidated Databricks & ADF examples and Readme into "Azure" dir
- Consolidated On Prem Readme and script into "OnPrem" dir

https://github.com/DTS-STN/CDP-Lessons-Learned/pull/13/commits/4c70f5d422ebb4ec01594cf66c3fa759b70ff04d

File: Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md (hunk 1 above)
Comment (JuntaoZ-NC) [reply Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md]: In this Commit, examples subdir has been modified :

- Consolidated Databricks & ADF examples and Readme into "Azure" dir
- Consolidated On Prem Readme and script into "OnPrem" dir

https://github.com/DTS-STN/CDP-Lessons-Learned/pull/13/commits/4c70f5d422ebb4ec01594cf66c3fa759b70ff04d

File: Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md (hunk 2 above)
Comment (JuntaoZ-NC) [reply Adobe-Azure-analytics-api-v2.0/examples/OnPrem/OnPremExample.md]: deleted readme and added description in this commit

https://github.com/DTS-STN/CDP-Lessons-Learned/pull/13/commits/4c70f5d422ebb4ec01594cf66c3fa759b70ff04d

**Comment by JuntaoZ-NC** [comment]
PR issues resolved and moved to https://github.com/DTS-STN/CDP-Lessons-Learned/pull/14

**Pull request closed without merging**